src/tracer.cc

```diff
@@ -369,30 +375,225 @@ TraceManager::Trace::InitTracer(
     }
   }
   exporter_ = otlp::OtlpHttpExporterFactory::Create(opts);
-  processor_ = otel_trace_sdk::SimpleSpanProcessorFactory::Create(
-      std::move(exporter_));
-  provider_ = otel_trace_sdk::TracerProviderFactory::Create(
-      std::move(processor_));
+  processor_ =
+      otel_trace_sdk::SimpleSpanProcessorFactory::Create(std::move(exporter_));
+  provider_ =
+      otel_trace_sdk::TracerProviderFactory::Create(std::move(processor_));
+  auto steady_timestamp_ns =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(
+          std::chrono::steady_clock::now().time_since_epoch())
+          .count();
+  auto otel_start_timestamp = otel_common::SystemTimestamp{
+      time_offset_ + std::chrono::nanoseconds{steady_timestamp_ns}};
+  auto root_span = InitSpan(
+      "InferRequest", otel_start_timestamp, steady_timestamp_ns,
+      true /*is_root_span*/);
+  // Initializing OTel context and storring "InferRequest" span as a root span
+  // to keep it alive for the duration of the request.
+  otel_context_ = opentelemetry::context::Context({kRootSpan, root_span});
```

Comment:
Should root span key also be appended with the request id?

Reply:
For a standalone model traces and ensembles, there can be only one root span "InferRequest" with the key "root_span", which shows timestamps of when request was received/send and when inference was complete. For now,  simple key without request id is sufficient. In future, it may change